core/app/models/spree/order_update_attributes.rb

```diff
@@ -0,0 +1,42 @@
+module Spree
+  class OrderUpdateAttributes
+    # @param order [Order] existing (persisted) order
+    # @param attributes [Hash] attributes which are assigned to the new order.
+    #   These attributes should already have been filtered.
+    #   * :payments_attributes attributes
+    def initialize(order, attributes, request_env: nil)
```

Comment:
Maybe that's the best way to do this, but I'm not sure. Can we discuss in another PR